changelog/2024-04-08T16_44_03+02_00_bitvector_add_ensurespine

```diff
@@ -0,0 +1 @@
+CHANGED: `BitVector n` now has an implementation for `ensureSpine` which ensures the constructor is present.
```

Comment:
I think this is a confusing statement.

The old implementation via generics already did this.
It would already transform `errorX ".." :: BitVector n` to `BV (errorX "..") (errorX "..")`.
Only because of the strict fields, that would turn right back into `errorX ".." :: BitVector n`.

The new implementation removes the `errorX` entirely and instead gives you `undefined#`, or in other words turns it into an BitVector with every bit undefined set to undefined/don't care.